database/vertica/vertica.go

```diff
@@ -67,21 +69,26 @@ func (v *Vertica) Open(url string) (database.Driver, error) {
 }
 
 func (v *Vertica) Close() error {
-	connErr := v.conn.Close()
-	dbErr := v.db.Close()
-	if connErr != nil || dbErr != nil {
-		return fmt.Errorf("conn: %v, db: %v", connErr, dbErr)
+	if e := v.conn.Close(); e != nil {
+		return fmt.Errorf("error closing vertica connection: %w", e)
+	}
+
+	if e := v.db.Close(); e != nil {
+		return fmt.Errorf("error closing vertica db: %w", e)
 	}
 	return nil
 }
 
+
+func (v *Vertica) dropSchema() error {
+	query := fmt.Sprintf("DROP SCHEMA %s CASCADE",  v.config.Schema)
+	_, e := v.conn.QueryContext(context.Background(), query)
+	return e
+}
+
+
 func (v *Vertica) dropTables() error {
-	// select all tables
-	whereStmt := ""
-	if len(v.config.Schema) > 0 {
-		whereStmt = ` WHERE table_schema='` + v.config.Schema + `'`
-	}
-	query := `SELECT DISTINCT table_name FROM tables` + whereStmt
+	query := `SELECT TABLE_NAME FROM v_catalog.tables`
```

Comment:
I think you still want the `WHERE table_schema=` clause, otherwise tables in other schemes will also be dropped. e.g. schemas that aren't managed by `migrate`

Reply:
That would be if the user does not specify a schema. There could be a use case where the migrations span the database in which drop should be scoped to the database. 
If a schema is specified it executes dropSchema()  which calls  `DROP SCHEMA %S CASCADE` which drops the entire schema. 

Dropping a database in vertica you have to stop vertica and delete files.  Destroying all the objects is the net affect. 


I am ok if you want to scope it to the schema but I feel if we do we should enforce that a schema is set.

Reply:
`DROP` is a dangerous operation, so it's better to be safe than sorry.
Let's use [CURRENT_SCHEMA](https://www.vertica.com/docs/9.2.x/HTML/Content/Authoring/SQLReferenceManual/Functions/VerticaFunctions/CURRENT_SCHEMA.htm) to get the current schema and limit our `DROP`s to that. I don't think requiring a schema to be specified makes sense if Vertica doesn't require it.